[ch155] As a user, I can see footer section

kimvcp opened the pull request:
https://app.clubhouse.io/cobda/story/155/as-a-user-i-can-see-footer-section

## Why?

This **PR** is for adding footer section.

## Changes

- Add a footer section containing `Logo`, `Language dropdown`, and other footer items. 
- Create another type of `Language dropdown`
- `Language dropdown` is not displayed when screen size is smaller than `md`
- Content supports both `TH` and `EN`.

## Screenshot(s)

**Before**

<img width="452" alt="Screen Shot 2564-02-24 at 21 00 43" src="https://user-images.githubusercontent.com/32285706/109011375-64756100-76e3-11eb-8a45-54fc32262798.png">

**After**

Mobile

<img width="448" alt="Screen Shot 2564-02-24 at 20 59 16" src="https://user-images.githubusercontent.com/32285706/109011169-355eef80-76e3-11eb-8e93-8630b76e843c.png">

Desktop

<img width="1300" alt="Screen Shot 2564-02-24 at 20 58 41" src="https://user-images.githubusercontent.com/32285706/109011176-3728b300-76e3-11eb-93e8-b2ad71bc8dc5.png">

kimvcp committed:
[ch155] Init simple footer layout

kimvcp committed:
[ch155] Split components into partials

kimvcp committed:
[ch155] Remove unnecessary passing prop

kimvcp committed:
[ch155] Adjust style for mobile responsive

kimvcp committed:
[ch155] Resolve conflits after rebase

kimvcp committed:
[ch155] Add icon for email box

kimvcp committed:
[ch155] Make language dropdown reusable

kimvcp committed:
[ch155] Modify text style for move responsive

kimvcp committed:
[ch155] Format code

kimvcp committed:
[ch155] Translate content

kimvcp committed:
[ch155] Fix default locale

kimvcp committed:
[ch155] Implement enum instead of boolean

Raksani commented on locales/th/common.json:
```diff
@@ -1,4 +1,11 @@
+  "helpCenter": "ศูนย์ให้บริการความช่วยเหลือ",
+  "termsOfService": "เงื่อนไขการให้บริการ",
+  "privacyPolicy": "นโยบายความเป็นส่วนตัว",
+  "emergencyLine": "สายฉุกเฉิน",
```
In the design, it uses “Emergency line” which actually means for local emergency line. For example, in [Thailand](https://en.m.wikipedia.org/wiki/Telephone_numbers_in_Thailand)

Would it be better to change it to be “Helpline”? since I saw people use the keyword, “Facebook helpline” to search for their telephone number. 

Or move it to`contact us: telephone` instead?

kimvcp replied:
Since we don't have any contact number yet, I think "Helpline" is a good fit for now.

Raksani commented on src/components/LanguageDropdown/index.tsx:
```diff
@@ -2,18 +2,26 @@ import React, { useEffect, useRef, useState } from 'react'
+
+const LanguageDropdown = ({ componentType }: LanguageDropdown) => {
+  const router = useRouter()
+  const { locales, locale } = router
```
Would you mind if I ask, is `locales` a plural form or a different one?

kimvcp replied:
`locales` in this case is the collection of all locales we have, but `locale` (without `s`) is the current one applying to our website. 
Further reading [here](https://nextjs.org/docs/api-reference/next/router)

This is the output:

<img width="164" alt="Screen Shot 2564-02-26 at 00 49 25" src="https://user-images.githubusercontent.com/32285706/109195540-3bc99600-77cd-11eb-886d-d91edcf87c20.png">

Raksani commented on src/styles/layouts/_layout-about-us.scss:
```diff
@@ -45,5 +45,9 @@
 
   & > footer {
     grid-area: footer;
+
+    display: flex;
+    justify-content: center;
+    box-shadow: 0 0 rem(20px) 0 box-shadow-opacity(0.16);
```
Can we use `@mixin` to make it reusable? since I saw you applied this pattern to many classes.

kimvcp replied:
Personally, I think we don't need to make the @mixin for this since there are many different `box-shadow` properties in our website.

LilBank commented on src/components/Footer/index.tsx:
```diff
@@ -1,15 +1,62 @@
+  const renderLogoColumn = () => (
+    <div className="footer__logo-container">
+      <Logo />
+      <LanguageDropdown componentType={ComponentType.Footer} />
```
`componentType` sounds strange; For example: `LanguageDropdown` has `componentType` `Footer`. What do you think about `childOf` or `parent` or any other meaningful names?

kimvcp replied:
I'm good to go with `parent`.

LilBank commented on src/components/Footer/index.tsx:
```diff
@@ -1,15 +1,62 @@
 import React from 'react'
 import Link from 'next/link'
+import Image from 'next/image'
+import Logo from '../Navbar/Logo'
```
How about move `Logo` out as an element instead of staying under `Navbar`?

LilBank commented on src/components/Footer/item.tsx:
```diff
@@ -0,0 +1,33 @@
+  readonly label: string
+  readonly href?: string
+  readonly isTitleText?: boolean
+  readonly isEmergencyText?: boolean
```
From `enum ComponentType`, we should create for these too. Like, `ContextType` or any other meaningful names?

LilBank commented on src/components/Footer/item.tsx:
```diff
@@ -0,0 +1,33 @@
+    isTitleText ? (
+      <h3 className="footer__list-title">{label}</h3>
+    ) : (
+      <Link href={href || '#'}>
```
Quick question, what does the hashtag refer to?

kimvcp replied:
At first I decided to use this `#` as the default one but seems like keeping it empty is a better idea.

LilBank commented on src/components/Footer/item.tsx:
```diff
@@ -0,0 +1,33 @@
+
+const FooterItem = ({ label, href, isTitleText, isEmergencyText }: FooterItem) => {
+  
+  const linkClassName: string = isEmergencyText
```
This could also be move into a function since it's not reusable.

kimvcp replied:
If we make it as a function, then we would have to store its value into variable and invoke it for returning string right? 
If so, what if we just store a string value into a variable and use it without invoking any function? 

Example code from `development` branch
<img width="681" alt="Screen Shot 2564-02-26 at 03 47 44" src="https://user-images.githubusercontent.com/32285706/109215188-645d8a00-77e5-11eb-9938-4290ad74ba92.png">

LilBank commented on src/components/LanguageDropdown/index.tsx:
```diff
@@ -2,18 +2,26 @@ import React, { useEffect, useRef, useState } from 'react'
 export enum LocaleCode {
   English = 'EN',
-  Thai = 'TH'
+  Thai = 'TH',
```
Intentional?
```suggestion
  Thai = 'TH'
```

LilBank commented on src/components/LanguageDropdown/index.tsx:
```diff
@@ -26,7 +34,9 @@ const NavbarLanguage = () => {
-    const isFocused: boolean | undefined = flagRef.current?.contains(event.target as Node)
+    const isFocused: boolean | undefined = flagRef.current?.contains(
+      event.target as Node
+    )
```
Could be in oneline I guess? Even if it's the formatter's work.

LilBank commented on src/components/LanguageDropdown/index.tsx:
```diff
@@ -46,15 +56,35 @@ const NavbarLanguage = () => {
 
+  const renderSelectedLocale = () => (
+    <>
+      <Image src="/icons/world.svg" height={16} width={16} />
```
Since the above `Image` tag is split into multiple lines, which are we sticking with? The multiple lines one?

kimvcp replied:
Single line looks better.

LilBank commented on src/components/Navbar/Tool.tsx:
```diff
@@ -1,12 +1,13 @@
 import NavbarAccount from './Account'
-import NavbarLanguage from './Language'
+import LanguageDropdown from '../LanguageDropdown'
+import { ComponentType } from '../Footer'
```
Should this enum become a global enum? There any many usage of this.

LilBank commented on src/styles/components/_dropdown.scss:
```diff
@@ -1,52 +1,78 @@
-    background-color: $light;
-    cursor: pointer;
-  }
+.dropdown-container {
```
Filename should match the css class. How about making the `dropdown-container` as an element within `dropdown`?

kimvcp committed:
[ch155] Format document

kimvcp committed:
[ch155] Rename dropdown element

kimvcp committed:
[ch155] Restructure files

kimvcp committed:
[ch155] Pass props enum instead of boolean

kimvcp committed:
[ch155] Resolve conflicts after rebase

kimvcp committed:
[ch155] Refine code for more readability

[ch155] Add type for a variable

[ch155] Remove unused css

[ch155] Remove unnecessary space

LilBank commented on src/styles/components/_footer.scss:
```diff
@@ -1,32 +1,102 @@
+  &__logo-container {
+    width: min-content;
+
+    & > * {
+      &:last-child {
+        display: none;
+
+        @include respond-above('md') {
+          display: block;
+          margin-top: rem(16px);
+        }
+
+        @include respond-above('lg') {
+          margin-top: rem(32px);
+        }
+      }
+    }
+  }
```
```suggestion
    .dropdown {
      display: none;

      @include respond-above('md') {
        display: block;
        margin-top: rem(16px);
      }

      @include respond-above('lg') {
        margin-top: rem(32px);
      }
    }
```

LilBank commented on src/styles/components/_footer.scss:
```diff
@@ -1,32 +1,102 @@
+    display: flex;
+    align-items: flex-end;
+
+    & > * {
+      &:last-child {
+        margin-left: rem(8px);
+      }
+    }
```
```suggestion
    .footer__list-item {
      margin-left: rem(8px);
    }
```

LilBank commented on src/styles/components/_dropdown.scss:
```diff
@@ -48,5 +62,17 @@
   &__label {
     margin-left: rem(8px);
+
+    &--bold {
+      font-weight: bold;
+    }
+
+    @include respond-above('md') {
+      font-size: rem(12px);
+    }
+
+    @include respond-above('lg') {
+      font-size: rem(16px);
+    }
```
```suggestion
    @include respond-above('md') {
      font-size: rem(12px);
    }

    @include respond-above('lg') {
      font-size: rem(16px);
    }

    &--bold {
      font-weight: bold;
    }
```

kimvcp committed:
[ch155] Refine css code
